eslint-plugin/rules/validate-imports.js

```diff
@@ -174,28 +174,25 @@ module.exports = {
           IMPORTED_FRAGMENTS = Object.create(null);
           IMPORT_LINE_USED = new Map();
 
-          const importLinesToInlinedSource = inlineImports.lineToImports(
-            context.getSourceCode().text,
-            {
-              resolveOptions: {
-                basedir: path.dirname(filename),
-              },
-              resolveImport,
-              throwIfImportNotFound: false,
-            },
-          );
+          const lineToFragmentNameToBucket = gatherFragmentImports({
+            source: context.getSourceCode().text,
+            sourceLocation: filename,
+            resolveImport,
+            fragmentParserGenerator: context.parserServices.getFragmentDefinitionsFromSource,
+            throwIfImportNotFound: false,
+          });
 
-          for (const [lineNumber, source] of importLinesToInlinedSource) {
+          for (const [lineNumber, fragmentNameToFragment] of Object.entries(
+            lineToFragmentNameToBucket,
+          )) {
             // initially we don't know that it's used
             // we'll mark the used ones as true
             // whatever false remains are unused
-            IMPORT_LINE_USED.set(lineNumber, false);
+            IMPORT_LINE_USED.set(Number(lineNumber), false);
```

Comment:
This change makes me think we should have `gatherFragmentImports` return a `Map<number,>`

Reply:
switched to map